src/poly/commitment.rs

```diff
@@ -154,6 +217,95 @@ impl<C: CurveAffine> Params<C> {
     }
 }
 
+/// A guard returned by the verifier
+#[derive(Debug)]
+pub struct Guard<'a, C: CurveAffine> {
+    /// Vector of random generators
+    pub g: Vec<C>,
+
+    /// Random generator
+    pub h: C,
+
+    /// Negation of z1 value in the OpeningProof
+    pub neg_z1: C::Scalar,
+
+    /// Params that were used by the verifier
+    pub params: &'a Params<C>,
+
+    /// Scalars produced by the verifier for multiscalar multiplication
+    pub scalars: Vec<C::Scalar>,
+
+    /// Points produced by the verifier for multiscalar multiplication
+    pub bases: Vec<C>,
+}
+
+impl<'a, C: CurveAffine> Guard<'a, C> {
+    /// Lets caller supply the challenges and obtain an MSM with updated
+    /// scalars and points.
+    pub fn use_challenges(
+        &mut self,
+        challenges_sq_packed: Vec<Challenge>,
+    ) -> Result<MSM<C>, Error> {
+        // - [z1] G
```

Comment:
This comment is out of position/context.

Reply:
(i've moved this comment down, not sure why the change isn't being reflected here)